packages/protocol/contracts/common/Accounts.sol

```diff
@@ -180,6 +195,59 @@ contract Accounts is
     emit AccountMetadataURLSet(msg.sender, metadataURL);
   }
 
+  /**
+   * @notice Set the indexed signer for a specific role
+   * @param signer the address to set as default
+   * @param role the role to register a default signer for
+   */
+  function setDefaultSigner(address signer, string memory role) public {
+    require(
+      isGenericSigner(msg.sender, signer, role),
+      "Must finish authorization before setting default signer"
+    );
+
+    Account storage account = accounts[msg.sender];
+    if (keccak256(abi.encodePacked(role)) == keccak256(abi.encodePacked(VoteSigner))) {
+      account.signers.vote = signer;
+    } else if (
+      keccak256(abi.encodePacked(role)) == keccak256(abi.encodePacked(AttestationSigner))
+    ) {
+      account.signers.attestation = signer;
+    } else if (keccak256(abi.encodePacked(role)) == keccak256(abi.encodePacked(ValidatorSigner))) {
+      account.signers.validator = signer;
+    }
+
+    account.defaultSigners[role] = signer;
+    emit DefaultSignerSet(msg.sender, signer, role);
+  }
+
+  /**
+   * @notice Authorizes an address to as a signer on behalf of the account.
+   * @param signer The address of the signing key to authorize.
+   * @param role The role to authorize signing for.
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
+   * @dev v, r, s constitute `signer`'s signature on `msg.sender`.
+   */
+  function authorizeSignerWithSignature(
+    address signer,
+    string memory role,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) public {
+    authorize(signer, v, r, s);
+
+    Account storage account = accounts[msg.sender];
+    account.signerAuthorizations[role][signer] = SignerAuthorization({
+      started: true,
+      completed: true
+    });
```

Comment:
I think these 2 booleans are packed into 1 storage slot so it only counts as 1 `SSTORE`

Reply:
yup